### PR TITLE
Add simple snippets support

### LINKED ISSUE #52: Add Snippets playground support
Add Playground or Snippets Support, so that people can quickly play with Markdown API

### PR DESCRIPTION
**Pull request by Kyle-Ye**
- Add simple snippets support
- Add .vscode to .gitignore file

Bug/issue #, if applicable: 

Close #52 

## Summary

Add Swift Snippets Support for swift-markdown package

See more discussion on this forum post https://forums.swift.org/t/swift-snippets/51947

## Dependencies

https://github.com/apple/swift-package-manager/pull/3694

## Testing
1. On VSCode

Navigate to the snippet file. Right-click to open the menu, choose `Swift` and then `Run Swift Snippet`

image

3. On Terminal
```shell
# checkout to this branch and cd into swift-markdown repo
export SWIFTPM_ENABLE_SNIPPETS=1 
swift package learn
```

```
# swift-markdown

## Products

- Markdown (library)
- markdown-tool (executable)

## Snippets

0. Parsing (2 snippets)
   Examples on how to use Swift Markdown Framework to parse the Markdown String into Swift object

1. Visiting (1 snippet)
   Examples on how to use Swift Markdown Framework to visit the Markdown object

Choose a group by name or number.
To exit, enter 'q'.
>>> 
```

```
# Parsing

Examples on how to use Swift Markdown Framework to parse the Markdown String into Swift object

0. ParseString
   Example on how to parse Markdown from string

1. ParseFile
   Example on how to parse Markdown from file

Choose a number or a name from the list of snippets.
To go back, press ↩.
To exit, enter `q`.
>>> 
```

## Checklist

- [x] Updated documentation if necessary

### CODE REVIEW

File: Snippets/Parsing/ParseFile.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//! Example on how to parse Markdown from file
+import Markdown
+import Foundation
+
+guard let url = Bundle.main.url(forResource: "Test", withExtension: "md") else {
```
Comment (Kyle-Ye) [inline Snippets/Parsing/ParseFile.swift:5]: This is not working as expected like Swift Playground. What's the best practice for adding a Snippest-only resource file and how are we supposed to get the resource? 

Could you help provide some infomation on this? cc @bitjammer

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by TimTr** [comment]
This is a little out of date, so wanted to follow-up. This looks like some snippets added to the markdown repo, at least in part to try out another feature in VSCode's Swift plugin. I'm not really seeing that functionality in VSCode, but may be missing it.

Related to this specific PR, though, is it still relevant? There are now quite a few snippets in the repo, and they are used within the DocC catalog as well. Should we just close this PR?

**Comment by Kyle-Ye** [comment]
Close since #94 is merged.

**Pull request closed without merging**